crates/router/src/types/api/webhooks.rs

```diff
@@ -109,7 +174,7 @@ pub trait IncomingWebhook: ConnectorCommon + Sync {
             .get_webhook_source_verification_signature(request)
             .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
         let secret = self
-            .get_webhook_source_verification_merchant_secret(db, merchant_id)
+            .get_webhook_source_verification_merchant_secret(db, merchant_id, connector_label)
```

Comment:
merchant key store is available in the parent function. Pass it on to this function and then use it in child functions instead of querying it again.